### DISCUSSION TITLE
Patch Patient - Error is:  body: must be a list of patch operation requests

### DISCUSSION THREAD

**Discussion by timammons** [FirelyTeam/firely-net-sdk#2217] [General]
I am trying to patch a patient in the following way, but getting an error **body: must be a list of patch operation requests**.  **_Any idea why?_**

**Code:**
`static void UpdatePatientName(FhirClient fireClient, string patientID)
{
    var pat = fireClient.Read<Patient>($"/Patient/{patientID}");
    var eTag = fireClient.LastResult.Etag;

    Parameters patParams = new Parameters();
    patParams.AddReplacePatchParameter("name/0/given", new HumanName().WithGiven("Beulah").WithGiven("Z"));

    fireClient.RequestHeaders.IfMatch.Add(new EntityTagHeaderValue($"\"{eTag}\"", true)); 
    var updatedPat = fireClient.Patch<Patient>($"{patientID}", patParams);
}`

**This is the body it created, which appears to be a list of patch operation requests:**
{"resourceType":"Parameters","parameter":[{"name":"operation","part":[{"name":"type","valueCode":"replace"},{"name":"path","valueString":"name/0/given"},{"name":"value","valueHumanName":{"given":["Beulah","Z"]}}]}]}

Thanks,
Tim

**Comment by ghost** [comment]
Could you post the full body that gets created ?

**Comment by timammons** [comment]
I am trying to patch a patient in the Cerner sandbox.
The PATCH request is:
PATCH /r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/Patient/12804999

The body being sent is:
```
{
	"resourceType": "Parameters",
	"parameter": [
		{
			"name": "operation",
			"part": [
				{
					"name": "type",
					"valueCode": "replace"
				},
				{
					"name": "path",
					"valueString": "name/0/given"
				},
				{
					"name": "value",
					"valueHumanName": {
						"given": [
							"Beulah",
							"Z"
						]
					}
				}
			]
		}
	]
}
```

Cerner documentation says they expect something more like this:

```
[
	{
		"path": "/name/0/id",
		"op": "test",
		"value": "12804999"
	},
	{
		"path": "/name/0/given",
		"op": "replace",
		"value": [
			"Beulah",
			"Z"
		]
	}
]
```

The C# code is in the first message I posted.  I'm trying to determine why my code and the SDK are producing what may be non-standard JSON.

Thanks,
Tim

**Comment by mbaltus** [comment]
The `AddReplacePatchParameter` will add a FHIRPath Patch parameter, following the rules that you can find here: http://www.hl7.org/fhir/fhirpatch.html. The FhirClient.Patch can then be used to request a FHIRPath Patch on the server. Cerner apparently does not use FHIRPath patch, but JSON Patch.

At the moment the SDK does not support JSON Patch. It has been requested in #1533. @marcovisserFurore: pinging you to maybe look into putting support for JSON Patch on the roadmap?

**Comment by timammons** [comment]
Mirjam,

Thank you for clearing that up.  I looked at the Cerner documentation, and they do indeed state that they follow the JSON PATCH spec.

- Tim

**Comment by ewoutkramer** [comment]
I have added this as an enhancement request to the SDK, but we have no direct plans to implement it. Though, as usual, we would welcome a PR ;-)
